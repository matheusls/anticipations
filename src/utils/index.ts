export const centsToReal = (cents: number) => {
  return new Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL',
  }).format(cents / 100);
};

export const realToCents = (real: string) => {
  return Number(real.replace(/\D+/g, ''));
};
