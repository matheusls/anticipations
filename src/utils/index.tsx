import { toast, TypeOptions } from 'react-toastify';

export const centsToReal = (cents: number) => {
  return new Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL',
  }).format(cents / 100);
};

export const realToCents = (real: string) => {
  return Number(real.replace(/\D+/g, ''));
};

type ErrorToastProps = {
  text: string;
  type: TypeOptions;
};

export const renderErrorToast = (error = '') => {
  const errorToastProps: ErrorToastProps = {
    '500': {
      text: 'Oops, tivemos um problema. Por favor, tente novamente.',
      type: 'error' as const,
    },
    '408': {
      text: 'Oops, nossos servidores estão demorando para responder...',
      type: 'warning' as const,
    },
    offline: {
      text: 'Você parece estar com problemas de conexão!',
      type: 'warning' as const,
    },
  }[error] || {
    text: 'Oops, houve um problema. Por favor, tente novamente.',
    type: 'error' as const,
  };

  const { text, type } = errorToastProps;

  return toast(text, {
    position: 'top-center',
    autoClose: false,
    closeOnClick: true,
    type,
  });
};
