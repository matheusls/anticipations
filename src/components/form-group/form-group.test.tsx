import { render } from 'tests/utils';

import FormGroup from './form-group';

describe('<FormGroup />', () => {
  it('should match snapshot', () => {
    const { container } = render(<FormGroup>FormGroup component</FormGroup>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
