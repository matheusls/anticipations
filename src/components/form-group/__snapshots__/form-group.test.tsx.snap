// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormGroup /> should match snapshot 1`] = `
.c0:not(:last-of-type) {
  margin-bottom: 26px;
}

<div
  class="c0"
>
  FormGroup component
</div>
`;
